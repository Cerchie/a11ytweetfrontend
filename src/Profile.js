import * as React from 'react'
import { useState, useContext } from 'react'
import UserContext from './UserContext'
import { useHistory } from 'react-router-dom'

function Profile({ saveProfile }) {
    if (useContext(UserContext) !== undefined) {
        const { currentUser } = useContext(UserContext)

        const [formData, setFormData] = useState({
            username: currentUser.username,
            password: '',
        })
        const [formErrors, setFormErrors] = useState([])
        const [saveConfirmed, setSaveConfirmed] = useState(false)

        console.debug(
            'ProfileForm',
            'currentUser=',
            currentUser,
            'formData=',
            formData,
            'formErrors=',
            formErrors,
            'saveConfirmed=',
            saveConfirmed
        )
        function goHome() {
            let history = useHistory()
            history.push('/gohome')
        }
        async function handleSubmit(evt) {
            evt.preventDefault()

            let profileData = {
                password: formData.password,
            }
            let username = formData.username
            let updatedUser

            try {
                updatedUser = await saveProfile(username, profileData)
            } catch (errors) {
                setFormErrors(errors)
                return
            }
            setFormData((f) => ({ ...f, password: '' }))
            setFormErrors([])
            setSaveConfirmed(true)
        }

        function handleChange(evt) {
            const { name, value } = evt.target
            setFormData((f) => ({
                ...f,
                [name]: value,
            }))
            setFormErrors([])
        }
        //maybe add a 'if Curruser...''
        return (
            <form onSubmit={handleSubmit}>
                <div className="shadow-md rounded-xl bg-blue-50 text-red-700 text-2xl p-6 m-6">
                    <>
                        <label
                            htmlFor="username"
                            className=" rounded-xl bg-white text-gray-600 text-2xl p-6 m-6"
                        >
                            Username
                        </label>
                        <input
                            id="username"
                            name="username"
                            className="form-control"
                            value={formData.username}
                            onChange={handleChange}
                            className="border p-2 m-6"
                        />
                    </>
                    <>
                        <label
                            htmlFor="password"
                            className=" text-red-700 text-2xl p-6 m-6"
                        >
                            Password
                        </label>
                        <input
                            id="password"
                            type="password"
                            name="password"
                            className="form-control"
                            value={formData.password}
                            onChange={handleChange}
                            className="border p-2 m-6"
                        />
                    </>
                    {saveConfirmed ? goHome() : null}
                    <button
                        type="submit"
                        className="btn btn-primary float-right"
                        onSubmit={handleSubmit}
                        className=" text-red-600 text-2xl"
                    >
                        Submit
                    </button>
                </div>
            </form>
        )
    }
    return <> No Profile </>
}

export default Profile
