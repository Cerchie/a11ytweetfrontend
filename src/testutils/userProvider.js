import React from 'react'
import UserContext from '../UserContext'

const demoUser = {
    username: 'testuser',
}

const UserProvider = ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
        {children}
    </UserContext.Provider>
)

export { UserProvider }
