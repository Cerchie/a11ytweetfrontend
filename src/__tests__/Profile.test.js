import { render, screen } from '@testing-library/react'
import React from 'react'
import Profile from '../Profile'
import { MemoryRouter } from 'react-router'
import { UserProvider } from '../testutils/userProvider'

test('renders comp', () => {
    render(<Profile />)
})
it('matches snapshot', function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <Profile />
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot()
})
