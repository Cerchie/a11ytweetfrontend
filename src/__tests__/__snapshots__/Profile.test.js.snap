// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div
      class="shadow-md rounded-xl bg-blue-50 text-red-700 text-2xl p-6 m-6"
    >
      <label
        class=" rounded-xl bg-white text-gray-600 text-2xl p-6 m-6"
        for="username"
      >
        Username
      </label>
      <input
        class="border p-2 m-6"
        id="username"
        name="username"
        value="testuser"
      />
      <label
        class=" text-red-700 text-2xl p-6 m-6"
        for="password"
      >
        Password
      </label>
      <input
        class="border p-2 m-6"
        id="password"
        name="password"
        type="password"
        value=""
      />
      <button
        class=" text-red-600 text-2xl"
        type="submit"
      >
        Submit
      </button>
    </div>
  </form>
</DocumentFragment>
`;
